iocore/hostdb/P_RefCountCache.h

```diff
@@ -290,9 +290,9 @@ RefCountCachePartition<C>::clear()
   // Hence, this monstrosity.
   auto it = this->item_map.begin();
   while (it != this->item_map.end()) {
-    auto cur = it;
+    auto cur = it++;
 
-    it = this->item_map.erase(it);
+    this->item_map.erase(cur);
```

Comment:
This change doesn't seem necessary.

Reply:
I think doing the post increment is cleaner style.

Reply:
At some point you may want to add a comment that erase() for intrusive hash map doesn't invalidate iterators, as this is different from std::unordered_map.

Reply:
It does invalidate the erased iterator. That's why you need `cur` and not just `it`.

Reply:
Yes obviously it invalidates the passed iterator.  But erase or add can invalidate _all_ iterators for some STL container types.  In particular, for std::unordered_map, I believe add and erase invalidate all iterators for a container if a rehash is triggered.

Reply:
For some containers, yes, but not for `map` / `unordered_map`. For instance, for `std::unordered_map::erase`,

> References and iterators to the erased elements are invalidated. Other iterators and references are not invalidated.

Therefore, this behaves the same as `std::unordered_map`.